File: lab-3-dynamodb/dynamoDBJavaLab/test/PatientReportLinkerTest.java
// Copyright 2017 Amazon Web Services, Inc. or its affiliates. All rights reserved.

import static org.junit.Assert.*;

import org.junit.Test;

public class PatientReportLinkerTest {

  @Test
  public void test() throws Exception {
    PatientReportLinker.main(new String[0]);
    for (int i = 1; i < 4; i++) {
      // Checks if PatientReportUrl attribute is present.
      boolean patientReportUrlAttrPresent =
          PatientReportLinker.dynamoDB
              .getTable(PatientReportLinker.INFECTIONS_TABLE_NAME)
              .getItem("PatientId", ("" + i))
              .isPresent("PatientReportUrl");
      System.out.printf(
          "Test - PatientId: %s, patientReportUrlAttrPresent: %b %n",
          i, patientReportUrlAttrPresent);
      assertTrue(patientReportUrlAttrPresent);
    }
  }
}

File: lab-3-dynamodb/dynamoDBPythonLab/patient_report_linker_test.py
# Copyright 2017 Amazon Web Services, Inc. or its affiliates. All rights
# reserved.

import unittest
import boto3
import warnings
import utils

TABLE_NAME = utils.LAB_S3_INFECTIONS_TABLE_NAME


class PatientReportLinkerTest(unittest.TestCase):

    def test_report_link(self):
        warnings.simplefilter("ignore", ResourceWarning)
        dynamodb = boto3.resource('dynamodb')
        myTable = dynamodb.Table(TABLE_NAME)
        for i in range(1, 4):
            isPresent = False
            rec = myTable.get_item(
                Key={'PatientId': str(i)})
            if rec.get('Item').get('PatientReportUrl'):
                isPresent = True
            self.assertTrue(isPresent)

if __name__ == '__main__':
    unittest.main()
